# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFT2FontList.h
# Commit: 7ae4df10749c
# Full Hash: 7ae4df10749c176009923591c252c634571c21d1
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-02 15:21:09
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r ece9a4223349 -r 7ae4df10749c gfx/thebes/gfxFT2FontList.h
--- a/gfx/thebes/gfxFT2FontList.h	Thu Jul 02 03:39:46 2020 +0000
+++ b/gfx/thebes/gfxFT2FontList.h	Thu Jul 02 02:51:41 2020 +0000
@@ -11,6 +11,9 @@
 #include "gfxPlatformFontList.h"
 
 namespace mozilla {
+
+class CacheAwareZipReader;
+
 namespace dom {
 class SystemFontListEntry;
 };
@@ -172,7 +175,7 @@
   void AppendFacesFromFontFile(const nsCString& aFileName,
                                FontNameCache* aCache, StandardFile aStdFile);
 
-  void AppendFacesFromOmnijarEntry(nsZipArchive* aReader,
+  void AppendFacesFromOmnijarEntry(mozilla::CacheAwareZipReader* aArchive,
                                    const nsCString& aEntryName,
                                    FontNameCache* aCache, bool aJarChanged);
 